include/seqan3/alphabet/quality/phred42.hpp

```diff
@@ -42,129 +42,151 @@ namespace seqan3
     {
-        return this->value <= rhs.value;
+        return this->_value <= rhs._value;
     }
 
-    constexpr bool operator>=(const illumina18 & rhs) const
+    constexpr bool operator>=(const phred42 & rhs) const
     {
-        return this->value >= rhs.value;
+        return this->_value >= rhs._value;
     }
     //!\}
 
-    //! explicit compatibility to char code of a quality score
+    /*!\name Conversion and explicit conversion functions.
+    * \{
+   */
+    //!\brief Explicit compatibility to char code of a quality score.
     explicit constexpr operator char() const
     {
         return to_char();
     }
 
-    //! convert quality score to its 1-letter code
+    //!\brief Convert quality score to its ascii representation.
     constexpr char_type to_char() const
     {
-        return static_cast<char_type>(value + offset_char);
+        return static_cast<char_type>(_value + offset_char);
     }
 
-    //! set internal value given 1-letter code
-    constexpr illumina18 & assign_char(char_type const c)
+    //!\brief Set internal value given its ascii representation.
+    constexpr phred42 & assign_char(char_type const c)
     {
-        value = char_to_value[c];
+        _value = char_to_value[c];
         return *this;
     }
 
-    //! explicit compatibility to internal rank representation
+    //!\brief Explicit compatibility to internal rank representation.
     constexpr rank_type to_rank() const
     {
-        return value;
+        return _value;
     }
 
-    //! set internal value given zero-based integer c
-    constexpr illumina18 & assign_rank(rank_type const c)
+    //!\brief Set internal value given phred integer code p.
+    constexpr phred42 & assign_phred(phred_type const p)
     {
-        assert(c < value_size);
-        value = c;
+        // p >= 0 is implicitly always true
+        assert(p < max_value_size);
+        _value = (p < value_size) ? p : value_size - 1;
         return *this;
     }
 
-    //! set internal value given Illumina 1.8 integer code p
-    constexpr illumina18 & assign_phred(phred_type const p)
+    //!\brief Set internal value given 0-based rank code p.
+    constexpr phred42 & assign_rank(phred_type const p)
     {
-        assert(p >= offset_phred && p < offset_phred + value_size);
-        value = p - offset_phred;
-        return *this;
+        return assign_phred(p);
     }
 
-    //! get Illumina 1.8 integer code
-    constexpr phred_type to_phred() const
+    //!\brief Return the integer rank code.
+    constexpr rank_type to_phred() const
     {
-        return value + offset_phred;
+        return _value;
     }
+    //!\}
 
-    //! phred score range for Illumina 1.8 standard
+    //!\brief The phred score range size for Illumina 1.8 standard.
     static constexpr rank_type value_size{42};
 
-protected:
+private:
+    //!\brief Maximally tolerated phred score [42..61] that will be mapped to 41.
+    static constexpr rank_type max_value_size{62};
```

Comment:
you can move this to the protected section also, but please mark everything in the protected section as \privatesection